### ISSUE TITLE
Issues with saving checkpoints

### ISSUE THREAD

**Issue by maevdokimov** [NVIDIA/NeMo#3181]
Hi! I'm trying to train tacotron 2 from scratch.

Running in current main branch with pytorch-lightning==1.5.0
```
PYTHONPATH="$(pwd)" python examples/tts/tacotron2.py \
train_dataset=train_data.json \
validation_datasets=validation_data.json \
+exp_manager.checkpoint_callback_params.save_top_k=1 \
+exp_manager.checkpoint_callback_params.monitor=loss
```

Results in correct training without checkpointing. Downgrading to pytorch-lightning==1.4.2 fixes this issue.

**Comment by blisc** [comment]
Is there any reason you are tracking training loss as opposed to validation loss? Do you run your experiments for enough epochs such that the validation happens at least once? Can you run with pytorch-lightning==1.5.1 as well?

Can you upload `lightning_logs.txt` from your experiment directory?
